### ISSUE TITLE
[Beatmap Editor] Slider outside of the working area

### ISSUE THREAD

**Issue by Rafrad** [ppy/osu#7598]
**Describe the crash:** 
Description: After the misplacement of slider osu! start to take a lot of processing power in Task Manager and crashes.
Computer System: macOS, Win10
Starting state: Editor compose screen
Steps:
-Select slider from the toolbox
-Put the first and second circle somewhere on the working area using left click
-Place the third circle:
   - very close to the second one so that their centres (white dots) partially overlap
   - at the white line symbolising distance between circles

Results: Slider creates almost straight line going out of the screen.
		 The computer is starting to heat up.
		 Osu starts to take a lot of processing power.
Note: The resources problem disappears only after quitting the game if it hasn't crashed.

**Screenshots or videos showing the encountered issue:** 
![Issue#3 Before crash](https://user-images.githubusercontent.com/48647578/73060640-92f98a80-3e98-11ea-828e-2395b07503f7.jpg)

Video
[https://drive.google.com/file/d/1l2Y2jEZVwJUP7jiH8waH9VfsISQGtKye/view?usp=sharing](url)

**osu!lazer version:** 
osu!lazer 2020.118.0

**Logs**

```
Logical CPU:     0
Error Code:      0x02000148
Trap Number:     133


Binary Images:
       0x10620e000 -        0x10621cfff +sh.ppy.osu.lazer (lazer - 0.0) <AB8D9C4F-4ED6-3AB1-B01A-EADF4BC87B98> /var/folders/*/osu! 2.app/Contents/MacOS/osu!
       0x10623d000 -        0x10628fff7 +libhostfxr.dylib (0) <2D58FB0F-67B4-3653-8F88-9C3ABC57DA22> /var/folders/*/osu! 2.app/Contents/MacOS/libhostfxr.dylib
       0x10629d000 -        0x1062e1fff +libhostpolicy.dylib (0) <BD0E2320-36F8-3A7F-838A-6A5CD93AB94D> /var/folders/*/osu! 2.app/Contents/MacOS/libhostpolicy.dylib
       0x106329000 -        0x106884fff +libcoreclr.dylib (0) <95F9BEEB-0267-3296-B5CA-39277DAD71D0> /var/folders/*/osu! 2.app/Contents/MacOS/libcoreclr.dylib
       0x106f72000 -        0x107197ff7 +libclrjit.dylib (0) <89D9B965-5563-39D4-9A20-893E725E1D49> /var/folders/*/osu! 2.app/Contents/MacOS/libclrjit.dylib
       0x107df6000 -        0x107dffffb +System.Native.dylib (0) <4442A93F-3D9D-385E-A5F7-244E77956B41> /var/folders/*/osu! 2.app/Contents/MacOS/System.Native.dylib
       0x107e49000 -        0x107e51fff +System.Globalization.Native.dylib (0) <C2386C23-0EEC-325E-9C89-073860C8B598> /var/folders/*/osu! 2.app/Contents/MacOS/System.Globalization.Native.dylib
       0x10ba83000 -        0x10baa9fff +libstbi.dylib (0) <E1BE96B3-F1A0-3CE7-8910-39F49A3BA3E8> /var/folders/*/osu! 2.app/Contents/MacOS/libstbi.dylib
       0x10bd8d000 -        0x10bd93ff7 +System.Security.Cryptography.Native.Apple.dylib (0) <76C52661-44D1-3A72-BC1F-087E626B09E5> /var/folders/*/osu! 2.app/Contents/MacOS/System.Security.Cryptography.Native.Apple.dylib
       0x10c3d1000 -        0x10c3defff  com.apple.iokit.IOHIDLib (2.0.0 - 2.0.0) <08CC730F-CFD2-328D-B338-43D4E801ED6B> /System/Library/Extensions/IOHIDFamily.kext/Contents/PlugIns/IOHIDLib.plugin/Contents/MacOS/IOHIDLib
```

**Computer Specifications:** 
  Model Name:	MacBook Air
  Model Identifier:	MacBookAir7,2
  Processor Name:	Intel Core i5
  Processor Speed:	1,8 GHz
  Number of Processors:	1
  Total Number of Cores:	2
  L2 Cache (per Core):	256 KB
  L3 Cache:	3 MB
  Hyper-Threading Technology:	Enabled
  Memory:	8 GB
  Boot ROM Version:	188.0.0.0.0
  SMC Version (system):	2.27f2

**Comment by bdach** [comment]
Most likely going to end up being a duplicate of #5538 (slightly different reproduction scenario, but the visual presentation has me thinking they've got to be closely related). Closing as such.
